### PR TITLE
Introduce secure security manager to project

### PR DESCRIPTION
**Pull request by jasontedor**
This commit migrates SecureSM, our secure security manager implementation, from its own repository to being a sub-project of Elasticsearch.

### CODE REVIEW

File: libs/secure-sm/build.gradle [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.elasticsearch.gradle.precommit.PrecommitTasks
+
+apply plugin: 'elasticsearch.build'
+apply plugin: 'nebula.optional-base'
```
Comment (rjernst) [inline libs/secure-sm/build.gradle]: Is this used?

**Review by rjernst — COMMENTED** [review]
This looks good, but I think there are 2 other changes necessary:
1. Modify the security policy to use the new jar name (I think that is why CI failed)
2. Modify `BootstrapForTesting` to make it work in intellij (see the line for plugin-classloader there)

File: libs/secure-sm/build.gradle (hunk 1 above)
Comment (jasontedor) [reply libs/secure-sm/build.gradle]: I removed this.

**Comment by jasontedor** [comment]
Thanks @rjernst. I force pushed a commit addressing your comments. I know we do not like force pushes in reviews but I force pushed here because we want to merge all the commits in this PR into master so as to preserve the original history of securesm. Adding more commits on top of my initial commit here would have a lot of other commits coming along for the ride that we do not want, we want only the original history from securesm and exactly one commit integrating securesm into elasticsearch.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
Closed via 1b3d529bef3cd7048b1cb021b1c280a7c5b7985b

**Pull request closed without merging**